dimod/include/dimod/quadratic_model.h

```diff
@@ -937,31 +939,37 @@ class QuadraticModel : public QuadraticModelBase<Bias, Index> {
     }
 
     index_type add_variable(Vartype vartype) {
-        bias_type lb;
-        bias_type ub;
         if (vartype == Vartype::BINARY) {
-            lb = 0;
-            ub = 1;
+            return this->add_variable(vartype, 0, 1);
         } else if (vartype == Vartype::SPIN) {
-            lb = -1;
-            ub = 1;
+            return this->add_variable(vartype, -1, +1);
         } else if (vartype == Vartype::INTEGER) {
-            ub = this->max_integer();
-            lb = 0;
+            return this->add_variable(vartype, 0, this->max_integer());
+        } else if (vartype == Vartype::REAL) {
+            return this->add_variable(vartype, 0, this->max_real());
         } else {
             throw std::logic_error("unknown vartype");
         }
-        return this->add_variable(vartype, lb, ub);
     }
 
     index_type add_variable(Vartype vartype, bias_type lb, bias_type ub) {
         assert(lb <= ub);
-        assert(lb >= -this->max_integer());
-        assert(ub <= this->max_integer());
+
+        assert(vartype != Vartype::BINARY || lb == 0);
+        assert(vartype != Vartype::BINARY || ub == 1);
+
+        assert(vartype != Vartype::SPIN || lb == -1);
+        assert(vartype != Vartype::SPIN || ub == +1);
```

Comment:
Seems redundant to require type _and_ bounds determined by the type.

Reply:
It's true, we could add a default version of this method.